// Copyright 2022 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message HealthcareAlphaDicomStore {
  string name = 1;
  HealthcareAlphaDicomStoreNotificationConfig notification_config = 2;
  map<string, string> labels = 3;
  string project = 4;
  string location = 5;
  string dataset = 6;
}

message HealthcareAlphaDicomStoreNotificationConfig {
  string pubsub_topic = 1;
}

message ApplyHealthcareAlphaDicomStoreRequest {
  HealthcareAlphaDicomStore resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteHealthcareAlphaDicomStoreRequest {
  string service_account_file = 1;
  HealthcareAlphaDicomStore resource = 2;
}

message ListHealthcareAlphaDicomStoreRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
  string Dataset = 4;
}

message ListHealthcareAlphaDicomStoreResponse {
  repeated HealthcareAlphaDicomStore items = 1;
}

service HealthcareAlphaDicomStoreService {
  rpc ApplyHealthcareAlphaDicomStore(ApplyHealthcareAlphaDicomStoreRequest) returns (HealthcareAlphaDicomStore);
  rpc DeleteHealthcareAlphaDicomStore(DeleteHealthcareAlphaDicomStoreRequest) returns (google.protobuf.Empty);
  rpc ListHealthcareAlphaDicomStore(ListHealthcareAlphaDicomStoreRequest) returns (ListHealthcareAlphaDicomStoreResponse);
}
